packages/react/src/jsx-runtime.tsx

```diff
@@ -18,11 +18,14 @@ type JSXFactory = (
   key?: string
 ) => ServerComponent
 
-export const jsxs: JSXFactory = (...args) => {
-  return {
-    ...jsx(...args),
-    $staticChildren: true,
-  }
+class StaticChildrenArray extends Array {}
```

Comment:
Is there better way to associate that information with an array? "children.$static = true"? "Object.defineProperty" & "non-enumerable"? 

Users can still write `children.map()` on the server, so our information should be opaque and should not affect results of `map`.

BTW, if you write `children.map()` on our `StaticChildrenArray`, it will covert to `Array` anyway which is good - children are no longer static.